libs/couch2pg/importer.js

```diff
@@ -0,0 +1,105 @@
+var _ = require('underscore'),
+    Promise = require('rsvp').Promise,
+    format = require('pg-format');
+
+var deleteDocuments = function(db, docIdsToDelete) {
+  if (docIdsToDelete && docIdsToDelete.length > 0) {
+    return db.query(
+      format('DELETE FROM couchdb WHERE doc->>\'_id\' in (%L)',
+        docIdsToDelete));
+  } else {
+    return Promise.resolve();
+  }
+};
+
+var storeSeq = function(db, seq) {
+  return db.query(format('UPDATE couchdb_progress SET seq = %L', seq));
+};
+
+/*
+NB: loadAndStore doesn't try to only load and store documents that have changed
+from postgres' perspective, because we presume if something appears in the
+changes feed it needs updating.
+
+If somehow this changes we can make this code more complicated.
+*/
+var loadAndStoreDocs = function(db, couchdb, concurrentDocLimit, docsToDownload) {
+  if (docsToDownload.length > 0) {
+    var changeSet = docsToDownload.splice(0, concurrentDocLimit);
+    var maxSeq = _.max(_.pluck(changeSet, 'seq'));
+
+    return couchdb.allDocs({
+      include_docs: true,
+      keys: _.pluck(changeSet, 'id')
+    }).then(function(couchDbResult) {
+      console.log('Pulled ' + couchDbResult.rows.length + ' results from couchdb');
+
+      console.log('Clearing any existing documents from postgresql');
+
+      return deleteDocuments(db, _.pluck(couchDbResult.rows, 'id'))
+        .then(function() {
+          return couchDbResult;
+        });
+    }).then(function(couchDbResult) {
+      console.log('Inserting ' + couchDbResult.rows.length + ' results into couchdb');
+
+      return db.query(format(
+        'INSERT INTO couchdb (doc) VALUES %L',
+        couchDbResult.rows.map(function(row) {
```

Comment:
I'm confused here. So in the previous function, we deleted all the couchDbResult doc ids as they were either updated or deleted.

Here, without any obvious filter I can see to differentiate delete from update, we are adding back all of the same doc ids.

I guess I'm not clear on how a deleted doc does not get added back. Shouldn't deleted docs be filtered out somewhere so that this only adds back updated docs? I'm probably missing something obvious.